sphinx_immaterial/cpp_domain_fixes.py

```diff
@@ -192,8 +201,15 @@ def describe_signature(
 This allows additional customizations of those functions to access the symbol.
 """
 
+DOMAIN_CLS_TYPE = Union[sphinx.domains.c.CDomain, sphinx.domains.cpp.CPPDomain]
```

Comment:
Maybe change to `AnyDomainType` or something else camel case since all uppercase is normally not used for types.

Reply:
Also this should be `Union[Type[sphinx.domains.c.CDomain], Type[sphinx.domains.cpp.CPPDomain]]`